yt/frontends/cholla/data_structures.py

```diff
@@ -42,23 +73,92 @@ def _detect_output_fields(self):
+        self.num_grids = np.prod(nprocs)
+
+        if self.num_grids > 1:
+            # When there's more than 1 grid, we expect the user to
+            # - have not changed the names of the output files
+            # - have passed the file written by process 0 to ``yt.load``
+            # Let's perform a sanity-check that self.index_filename has the
+            # expected suffix for a file written by mpi-process 0
+            if int(_split_fname_proc_suffix(self.index_filename)[1]) != 0:
+                raise ValueError(
+                    "the primary file associated with a "
+                    "distributed cholla dataset must end in '.0'"
+                )
 
     def _parse_index(self):
-        self.grid_left_edge[0][:] = self.ds.domain_left_edge[:]
-        self.grid_right_edge[0][:] = self.ds.domain_right_edge[:]
-        self.grid_dimensions[0][:] = self.ds.domain_dimensions[:]
-        self.grid_particle_count[0][0] = 0
-        self.grid_levels[0][0] = 0
+        self.grids = np.empty(self.num_grids, dtype="object")
+
+        # construct an iterable over the pairs of grid-index and corresponding
+        # filename
+        if self.num_grids == 1:
+            ind_fname_pairs = [(0, self.index_filename)]
+        else:
+            # index_fname should has the form f'{self.directory}/<prefix>.0'
+            # strip off the '.0' and determine the contents of <prefix>
+            pref, suf = _split_fname_proc_suffix(self.index_filename)
+            assert int(suf) == 0  # sanity check!
+
+            ind_fname_pairs = ((i, f"{pref}.{i}") for i in range(self.num_grids))
+
+        dims_global = self.ds.domain_dimensions[:]
+        pbar = get_pbar("Parsing Hierarchy", self.num_grids)
+
+        # It would be nice if we could avoid reading in every hdf5 file during
+        # this step... (to do this, Cholla could probably encode how the blocks
+        # are sorted in an hdf5 attribute)
+
+        for i, fname in ind_fname_pairs:
+            if self.num_grids == 1:
+                # if the file was concatenated, we might be missing attributes
+                # that are accessed in the other branch. To avoid issues, we use
+                # hardcoded values
+                left_frac, right_frac, dims_local = 0.0, 1.0, dims_global
+            else:
+                with h5py.File(fname, "r") as f:
+                    offset = f.attrs["offset"][:].astype("=i8")
+                    dims_local = f.attrs["dims_local"][:].astype("=i8")
+                left_frac = offset / dims_global
+                right_frac = (offset + dims_local) / dims_global
+
+            level = 0
+
+            self.grids[i] = self.grid(
+                i,
+                index=self,
+                level=level,
+                dims=dims_local,
+                filename=fname,
+            )
+
+            self.grid_left_edge[i] = left_frac
+            self.grid_right_edge[i] = right_frac
+            self.grid_dimensions[i] = dims_local
```

Comment:
```suggestion
            self.grid_left_edge[i,:] = left_frac
            self.grid_right_edge[i,:] = right_frac
            self.grid_dimensions[i,:] = dims_local
```

Just for clarity, could we make it obvious that it's setting a slice to the values?